File: child-dir/src/main/java/codereview/simpleorder/order/dto/event/OrderCreatedEvent.java
package codereview.simpleorder.order.dto.event;

import codereview.simpleorder.order.domain.OrderItem;

import java.util.List;
import java.util.Map;

import static java.util.stream.Collectors.toMap;

public class OrderCreatedEvent {

    private final Map<Long, Integer> itemIdToDecreaseQuantity;

    public OrderCreatedEvent(List<OrderItem> orderItems) {

        this.itemIdToDecreaseQuantity = createInnerDto(orderItems);
    }

    private static Map<Long, Integer> createInnerDto(List<OrderItem> orderItems) {

        return orderItems.stream()
                .collect(toMap(OrderItem::getItemId, OrderItem::getOrderItemQuantity));
    }

    public Map<Long, Integer> values() {

        return itemIdToDecreaseQuantity;
    }
}

File: order/src/main/kotlin/msa/with/ddd/order/dto/event/OrderCreatedEvent.kt
package msa.with.ddd.order.dto.event

import msa.with.ddd.order.domain.OrderItem
import java.util.stream.Collectors.toMap

class OrderCreatedEvent(orderItems: List<OrderItem>) {

    private val itemIdToDecreaseQuantity: Map<Long, Int> = createInnerDto(orderItems)

    fun values(): Map<Long, Int> {
        return itemIdToDecreaseQuantity
    }

    companion object {
        private fun createInnerDto(orderItems: List<OrderItem>): Map<Long, Int> {
            return orderItems.stream()
                .collect(toMap({ it.itemId }, { it.orderItemQuantity }))
        }
    }
}